source/code-snippets/transactions/txn-convenient.js

```diff
@@ -0,0 +1,81 @@
+import { MongoError, MongoClient, ObjectId } from "mongodb";
+
+const uri =
+  "<connection string uri>";
+const client = new MongoClient(uri);
+await client.connect();
+
+await client
+  .db("testdb")
+  .collection("inventory")
+  .insertMany([
+    { item: "sunblock, 8 oz", qty: 85, price: 6.0 },
+    { item: "beach chair", qty: 30, price: 25.0 },
+  ]);
+
+// start-order-successful
+const order1 = [
+  { item: "sunblock, 8 oz", qty: 3 },
+  { item: "beach chair", qty: 1 }
+];
+// end-order-successful
+
+// start-order-fail
+const order2 = [
+  { item: "volleyball", qty: 1 }
+];
+// end-order-fail
+
+// start-transaction
+const txnResult = await client.withSession(async (session) =>
+  session
+    .withTransaction(async () => {
```

Comment:
```suggestion
    .withTransaction(async (session) => {
```

Minor, I think there's actually no functional difference but technically the API is that you should use the session passed to the transaction function, and shadowing variables like this is a perfectly fine way to make sure you don't grab something of the same name from the outer scope.